fv3core/stencils/neg_adj3.py

```diff
@@ -299,70 +298,80 @@ def compute(qvapor, qliquid, qrain, qsnow, qice, qgraupel, qcld, pt, delp, delz,
+        self._sum2 = utils.make_storage_from_shape(shape_ij, origin=(0, 0))
+        self._upper_fix = utils.make_storage_from_shape(qvapor.shape, origin=(0, 0, 0))
+        self._lower_fix = utils.make_storage_from_shape(qvapor.shape, origin=(0, 0, 0))
+        if namelist.check_negative:
+            raise NotImplementedError(
+                "Unimplemented namelist value check_negative=True"
+            )
+        if namelist.hydrostatic:
+            self._d0_vap = constants.CP_VAP - constants.C_LIQ
+            raise NotImplementedError("Unimplemented namelist hydrostatic=True")
+        else:
+            self._d0_vap = constants.CV_VAP - constants.C_LIQ
+        self._lv00 = constants.HLV - self._d0_vap * constants.TICE
+
+        self._fix_neg_water = StencilWrapper(
+            func=fix_neg_water,
+            origin=grid.compute_origin(),
+            domain=grid.domain_shape_compute(),
+        )
+        self._fillq = StencilWrapper(
+            func=fillq,
+            origin=grid.compute_origin(),
+            domain=grid.domain_shape_compute(),
+        )
+        self._fix_water_vapor_down = StencilWrapper(
+            func=fix_water_vapor_down,
+            origin=grid.compute_origin(),
+            domain=grid.domain_shape_compute(),
+        )
+        self._fix_neg_cloud = StencilWrapper(
+            func=fix_neg_cloud,
+            origin=grid.compute_origin(),
+            domain=grid.domain_shape_compute(),
+        )
+
+    def __call__(
+        self,
         qvapor,
         qliquid,
         qrain,
         qsnow,
         qice,
         qgraupel,
-        lv00,
-        d0_vap,
-        origin=grid.compute_origin(),
-        domain=grid.domain_shape_compute(),
-    )
-    fillq(
-        qgraupel,
-        delp,
-        sum1,
-        sum2,
-        origin=grid.compute_origin(),
-        domain=grid.domain_shape_compute(),
-    )
-    fillq(
-        qrain,
-        delp,
-        sum1,
-        sum2,
-        origin=grid.compute_origin(),
-        domain=grid.domain_shape_compute(),
-    )
-
-    fix_water_vapor_down(
-        qvapor,
+        qcld,
+        pt,
         delp,
-        upper_fix,
-        lower_fix,
-        origin=grid.compute_origin(),
-        domain=grid.domain_shape_compute(),
-    )
-
-    fix_neg_cloud(
-        delp, qcld, origin=grid.compute_origin(), domain=grid.domain_shape_compute()
-    )
+        delz,
+        peln,
+    ):
+        self._fix_neg_water(
+            pt,
+            delp,
+            delz,
+            qvapor,
+            qliquid,
+            qrain,
+            qsnow,
+            qice,
+            qgraupel,
+            self._lv00,
+            self._d0_vap,
+        )
+        self._fillq(qgraupel, delp, self._sum1, self._sum2)
+        self._fillq(qrain, delp, self._sum1, self._sum2)
+        self._fix_water_vapor_down(qvapor, delp, self._upper_fix, self._lower_fix)
+        self._fix_neg_cloud(delp, qcld)
```

Comment:
These could be all one stencil. Chris Kung I think had previously refactored it to be so but we were worried about losing the physical meaning so never merged it. No need to change it here, but something to keep in mind if we need to reduce the stencil call count, these can be joined.

Reply:
Well to keep the physical meaning you can keep that structure as `@gtstencil.function` into a merged stencil

Reply:
no, you can't have with computations in a function. you need separate computations in this algorithm.